osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -113,6 +126,16 @@ private void load()
                 throw new InvalidOperationException($"{GetType().ReadableName()} must provide a {nameof(JudgementResult)} through {nameof(CreateResult)}.");
 
             loadSamples();
+
+            userPositionalHitSounds = config.GetBindable<bool>(OsuSetting.PositionalHitSounds);
+            userPositionalHitSounds.BindValueChanged(positional =>
+            {
+                if (positional.NewValue)
+                    Samples?.AddAdjustment(AdjustableProperty.Balance, positionalSoundAdjustment);
```

Comment:
rather than adding and removing the adjustment, *always* bind the adjustment and then set the valule differently based on the enable/disable (ie. set to default balance if user does not want positional).